pulse-visualizer: init at 1.2.2

fndov opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fndov committed:
pulse-visualizer: init at 1.0.1

Append to maintainers list

Append to maintainers list V2

Append new line to package

Rebase

Apply suggested changes

Update platforms

Co-authored-by: Yiyu Zhou <yiyuzhou19@gmail.com>

Update platforms V2

Co-authored-by: Yiyu Zhou <yiyuzhou19@gmail.com>

pulse-visualizer: init at 1.2.2

Fix minor issue with version number, extra v

fndov commented:
@NixOS/nix-formatting

yzhou216 commented on pkgs/by-name/pu/pulse-visualizer/package.nix:
```diff
@@ -0,0 +1,72 @@
+  src = fetchFromGitHub {
+    owner = "Beacroxx";
+    repo = "pulse-visualizer";
+    rev = "882d6072f651ae00fcba66c9c2d5df2835b6cf61";
```
```suggestion
    tag = "v${finalAttrs.version}";
```

yzhou216 commented on pkgs/by-name/pu/pulse-visualizer/package.nix:
```diff
@@ -0,0 +1,72 @@
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "pulse-visualizer";
+  version = "1.0.1";
```
Please bump to upstream [1.1.0](https://github.com/Beacroxx/pulse-visualizer/releases/tag/v1.1.0).

yzhou216 commented on pkgs/by-name/pu/pulse-visualizer/package.nix:
```diff
@@ -0,0 +1,72 @@
+
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace-warn " -march=native" "" \
+      --replace-warn " -mtune=native" "" \
+      --replace-warn "-Wl,-s" "" \
+      --replace-warn " -s" "" \
+      --replace-warn 'set(CMAKE_INSTALL_PREFIX "/usr" CACHE PATH "Installation prefix" FORCE)' ""
```
```suggestion
      --replace-fail " -march=native" "" \
      --replace-fail " -mtune=native" "" \
      --replace-fail "-Wl,-s" "" \
      --replace-fail " -s" "" \
      --replace-fail 'set(CMAKE_INSTALL_PREFIX "/usr" CACHE PATH "Installation prefix" FORCE)' ""
```

`--replace-fail` is more robust use here.

yzhou216 reviewed (commented):
Welcome!

yzhou216 commented:
> @NixOS Can I have help with the formatting error?

Did you try formatting with `nixfmt`?

yzhou216 commented on maintainers/maintainer-list.nix:
This should be a separate commit as `maintainers: add miyu`.

yzhou216 replied:
This is why the CI is failing.

fndov commented:
Okay, I've been added to the maintainers list in a separate pull request, and I've updated the build with everything suggested, I've rebased to make it one commit, this is ready now.

fndov commented:
That's unfortunate, can you guide me to make the package x86_64 exclusive?

yzhou216 reviewed (commented):
> That's unfortunate, can you guide me to make the package x86_64 exclusive?

Sure

yzhou216 commented on pkgs/by-name/pu/pulse-visualizer/package.nix:
```diff
@@ -0,0 +1,70 @@
+    homepage = "https://github.com/Beacroxx/pulse-visualizer";
+    license = lib.licenses.gpl3;
+    maintainers = with lib.maintainers; [ miyu ];
+    platforms = lib.platforms.linux;
+    badPlatforms = [ "aarch64-linux" ];
```
```suggestion
    platforms = lib.platforms.x86_64;
    badPlatforms = lib.platforms.darwin;
```

Sorry, actually I think this might be a better choice since pipewire does work on other POSIX systems such as FreeBSD.

yzhou216 reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.

infinisil reviewed (approved):
That's pretty cool, sorry for the delay, let's merge!
